### ISSUE TITLE
further_reading with esm_master

### ISSUE THREAD

**Issue by JanStreffing** [esm-tools/esm_tools#1085]
For the new oifs48r1 I want a separate oifs.env.yaml, which I called oifs48.env.yaml
At first I tried to place the different further reading blocks into the compile_infos like so:
```
compile_infos:
        choose_version:
          48r1:
            branch: main
            comp_command: "./openifs-bundle build --threads=32 --arch=esm_tools --openifs-only --without-double-precision --install; chmod -R ${source_code_permissions} .; mv install/bin/ifsMASTER.SP install/bin/oifs"
            clean_command: ./openifs-bundle build --clean
            git-repository: https://gitlab.dkrz.de/ec-earth/oifs-48r1.git
            install_bins: install/bin/oifs
            destination: oifs-48r1
            further_reading:
              - oifs/oifs48.env.yaml
```
Since this did not seem to do anything I instead placed it in the version choose block:
```
choose_version:
        48r1:
                further_reading:
                        - oifs/oifs48.env.yaml
```
Again no luck.
I'm trying to get these variables into the compile script:

```
# Setup up compile and runtime environment i.e. what needs
# to be changed w.r.t the default config/machines/<computer>.yaml
compiletime_environment_changes:
  choose_computer.name:
    juwels:
      add_export_vars:
        - OIFS_CYCLE: 48r1
        - OIFS_HOME: "${model_dir}"
        - OIFS_LOGFILE: "$OIFS_HOME/oifs_test_log.txt"
        - OIFS_DATA_DIR: "/p/project/chhb19/streffing1/input/oifs-48r1/"

environment_changes:
  OpenMP_flag: ""
  OpenMP_CCDEFS: ""
```

You can find my attempts at:
https://github.com/esm-tools/esm_tools/tree/oifs48r1

**Comment by nwieters** [comment]
Hi @JanStreffing,

I am not sure if this solves the problem. But I found that the entries in `add_export_vars` for juwels (in e.g. `configs/components/oifs/oifs.env.yaml`) are placed between ''.

```
compiletime_environment_changes:
  choose_computer.name:
    juwels:
      add_export_vars:
        - 'OIFS_CYCLE: 48r1'
        - 'OIFS_HOME: "${model_dir}"'
        - 'OIFS_LOGFILE: "$OIFS_HOME/oifs_test_log.txt"'
        - 'OIFS_DATA_DIR: "/p/project/chhb19/streffing1/input/oifs-48r1/"'

```

**Comment by mandresm** [comment]
HI @JanStreffing,

There is a syntax problem in your `add_export_vars`, you are mixing the dictionary syntax and the list syntax. As you have it ESM-Tools will write the following in the bash scripts:

```bash
export OIFS_CYCLE: 48r1
export OIFS_HOME: "<dir_to_awicm3>"
export OIFS_LOGFILE: "$OIFS_HOME/oifs_test_log.txt"
export OIFS_DATA_DIR: "/p/project/chhb19/streffing1/input/oifs-48r1/"
```

Please, for anything that is new use the dictionary syntax for the `export_vars`:

```yaml
compiletime_environment_changes:
  choose_computer.name:
    juwels:
      add_export_vars:
        OIFS_CYCLE: 48r1
        OIFS_HOME: "${model_dir}"
        OIFS_LOGFILE: "$OIFS_HOME/oifs_test_log.txt"
        OIFS_DATA_DIR: "/p/project/chhb19/streffing1/input/oifs-48r1/"
```

**Comment by mandresm** [comment]
Hi @JanStreffing, is it solved? When I replied I made a mistake on my answer. Then I corrected it, but I'm guessing you don't get a correction email. I suggest you look at the issue on GitHub since that one is the corrected one, in case you are going to be trying to solve this problem you're having soon.

**Comment by JanStreffing** [comment]
Pardon, I did not see the last message until now. I'll check later

**Comment by JanStreffing** [comment]
This does not seem to make a difference. In general I think the issue lies not with what's written inside the oifs48.env.yaml, but rather with the it not being evaluated.

**Comment by JanStreffing** [comment]
Wait, nvm, I'm on levante now, not juwels. i need a new block inside

**Comment by JanStreffing** [comment]
Still no OIFS_CYCLE export.. I pushed the current state to https://github.com/esm-tools/esm_tools/tree/oifs48r1

**Comment by mandresm** [comment]
I see. `further_readings` are not resolved at the time the environment tries to access them. We would need to include them as early choose-resolved variables to fix the problem.

In the mean time I recommend to put the `further_reading` at the 1st nesting level (out of the `choose`) so that you can move forward. I'll try to have a look for the solution for the early choose resolution tomorrow.

**Comment by mandresm** [comment]
Could not look at the problem yet

**Comment by nwieters** [comment]
There are entries for `add_further_reading` in setups/awicm3/awicm3.yaml inside a choose_version: block. 
But unfortunately this doesn't work inside oifs.yaml. 

Did `add_further_reading` worked at some point?

(In awicm3.yaml it wants to add xios.env.yaml that only makes changes for computer ollie.)

Maybe we need to reactivate the funtionality for `add_further_reading`.

**Comment by nwieters** [comment]
This is now implemented by the PR #1096 and #1097.
